docs/federation.md

```diff
@@ -0,0 +1,126 @@
+## Federation
+
+AMKO has a federator which can federate the AMKO configuration to a set of member clusters. This is done to ensure that a user can initiate a recovery whenever there's some kind of a disaster (without the need of backing up the AMKO configs).
+
+### Federation Set
+Federation works on a set of clusters where AMKO is deployed. We will call this set of clusters as the federation set. One of these clusters in the federation set must be designated as the **leader**. And, all the other clusters are designated as **followers**.
+
+### Responsibilities of a leader and a follower
+1. A **leader** cluster is responsible for distributing the AMKO configuration to all the follower clusters in the federation set. AMKO in the leader cluster is also responsible to create/update GSLB Services on the Avi Controller.
+2. AMKO in a **follower** cluster runs in a no-op mode. The federator in the follower clusters won't federate the objects. And, AMKO in the follower clusters won't sync the GSLB Services in the Avi Controller. It just runs in a passive mode.
```

Comment:
AMKO in a **follower** cluster runs passively and does not carry out federation activities or GslbService sync operations unlike the leader.